x/gobenchdata-web/web/app.js

```diff
@@ -55,110 +58,134 @@ export async function generateCharts({
-        });
-
-        // attach group to parent
-        group.appendChild(document.createElement('hr'));
-        group.appendChild(document.createElement('br'));
-        div.appendChild(group);
+          // chart for each benchmark type
+          let seedColor = randomInt();
+
+          let benchmarks = [bench];
+
+          chartsTypes.forEach(c => {
+            const chartName = makeChartName(configCharName, c);
+
+            // create elements
+            const canvas = document.createElement('canvas');
+            canvas.id = configCharName;
+            const ctx = canvas.getContext('2d');
+
+            // create chart
+            let i = seedColor;
+            let max = 0;
+
+            const datasets = benchmarks.map(bench => {
+              const p = newRunPoint(c, run, bench);
+              max = Math.max(p.y + p.y * 0.1, max);
+              i += 3;
+              return newDataset(bench.Name, i, [p]);
+            });
+
+            charts[chartName] = new Chart(ctx, {
+              type: 'line-with-guides',
+              data: {
+                labels,
+                datasets,
+              },
+              options: chartOptions(c, max),
+            });
+
+
+            console.log('NEW CHART', datasets.length, charts[chartName].data.datasets, benchmarks.length);
+            // TODO: this only works if you click on a point exactly, which is
+            // dumb. can't seem to make it work for clicking anywhere (getting
+            // the chart.js x-axis is nontrivial). ugh
+            canvas.onclick = (e) => {
+              const p = charts[chartName].getElementAtEvent(e);
+              if (p && p.length) {
+                const { _index: i, _xScale: x } = p[0];
+                const label = x.ticks[i].split(' ');
+                // a commit label should have 2 parts (see label())
+                if (label.length === 1) return;
+                window.open(`https://${source}/commit/${label[0]}`, '_blank');
+              }
+            }
+
+            // attach to dom
+            const canvasDiv = document.createElement('div');
+            canvasDiv.setAttribute('class', 'canvaswrapper');
+            canvasDiv.appendChild(canvas);
+            group.appendChild(canvasDiv);
+          });
+
+          // attach group to parent
+          group.appendChild(document.createElement('hr'));
+          group.appendChild(document.createElement('br'));
+          div.appendChild(group);
+
+          // In the event that we're adding graphs per package and not per benchmark we need to break so we're not
+          // creating a new chart for each benchmark.
+          if(!perBenchmark){
+            break;
+          }
+        }
       }
     });
-
     // fill missing data from datasets
     Object.values(charts).forEach(c => {
-      const { data: { datasets } } = c;
+      const {data: {datasets}} = c;
       datasets.forEach(d => {
-        const { data } = d;
+        const {data} = d;
```

Comment:
nit: generally try to avoid style-only changes in PRs, though granted I never set up a linter so don't worry about it this time 😅